src/lzf.cc

```diff
@@ -17,75 +17,70 @@ using namespace node;
 
 
 // Handle<Value> ThrowNodeError(const char* what = NULL) {
-//     return NanThrowError(Exception::Error(NanNew<String>(what)));
+//     return Nan::ThrowError(Exception::Error(Nan::New<String>(what)));
 // }
 NAN_METHOD(compress) {
-    if (args.Length() < 1 || !Buffer::HasInstance(args[0])) {
-        return NanThrowError("First argument must be a Buffer");
+    if (info.Length() < 1 || !Buffer::HasInstance(info[0])) {
+        return Nan::ThrowError("First argument must be a Buffer");
     }
-    NanScope();
 
-    Local<Object> bufferIn = args[0]->ToObject();
+    Local<Object> bufferIn = info[0]->ToObject();
     size_t bytesIn         = Buffer::Length(bufferIn);
     char * dataPointer     = Buffer::Data(bufferIn);
     size_t bytesCompressed = bytesIn + 100;
     char * bufferOut       = (char*) malloc(bytesCompressed);
 
     if (!bufferOut) {
-        return NanThrowError("LZF malloc failed!");
+        return Nan::ThrowError("LZF malloc failed!");
     }
 
     unsigned result = lzf_compress(dataPointer, bytesIn, bufferOut, bytesCompressed);
 
     if (!result) {
         free(bufferOut);
-        return NanThrowError("Compression failed, probably too small buffer");
+        return Nan::ThrowError("Compression failed, probably too small buffer");
     }
 
-    Local<Object> BufferOut = NanNewBufferHandle(bufferOut, result);
-    free(bufferOut);
+    Nan::MaybeLocal<Object> BufferOut = Nan::NewBuffer(bufferOut, result);
```

Comment:
Will `Nan::NewBuffer()` internally call `realloc()` for the buffer? If not, it should be called before `Nan::NewBuffer()`, because now the code always returns `bytesIn + 100` bytes (with `result` bytes stored in it).